types/handlebars/index.d.ts

```diff
@@ -1,15 +1,48 @@
-// Type definitions for Handlebars v4.0.5
+// Type definitions for Handlebars v4.0.11
 // Project: http://handlebarsjs.com/
-// Definitions by: Boris Yankov <https://github.com/borisyankov>
+// Definitions by: Boris Yankov <https://github.com/borisyankov>, Sergei Dorogin <https://github.com/evil-shrike>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
 declare namespace Handlebars {
-    export function registerHelper(name: string, fn: Function, inverse?: boolean): void;
-    export function registerHelper(name: Object): void;
-    export function registerPartial(name: string, str: any): void;
+    export interface TemplateDelegate<T = any> {
+        (context: T, options?: RuntimeOptions): string;
+    }
+
+    export type Template<T = any> = TemplateDelegate<T>|string;
```

Comment:
Whats the use of declaring type parameter T if you arent using it anywhere in the definitions..  ?

Reply:
@sheetalkamat 
well, honestly I don't see much sense in type argument here, but it was there before so I didn't want to break definition.
strictly speaking T does used in the definition:
```
    export interface TemplateDelegate<T = any> {
        (context: T, options?: RuntimeOptions): string;
    }
```
so you can get a typed template:
```
let tmpl = template<MyType>(source);
// ok
tmpl(new MyType());
// error
tmpl({});
```

Reply:
@sheetalkamat ping

Reply:
what i am asking is are the usages in HelperOptions and others using these template and template delegates will always be any? Do you need type parameter in there?

Reply:
sorry, not sure I understand the point, I provided an example above where you specify a type for template and it restricts usage of that template to instances of that type. So it makes sense for TemplateDelegate. For HelperOptions and HelperDelegate it's a bit different picture. You don't use helpers directly, Handlebars calls them. So specifying a type for context has less sense but possible
```
Handlebars.registerHelper("xxx", (context, options) => {
  return "";
});
```
Are you suggesting adding a T to registerHelper?